src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java

```diff
@@ -417,48 +418,48 @@ private static void validateAllCheckTokensAreReferencedInConfigFile(String confi
                 throw new CheckstyleException("Couldn't find check: " + checkName, ex);
             }
 
-            if (instance instanceof AbstractCheck) {
-                final AbstractCheck check = (AbstractCheck) instance;
-                if (isAllTokensAcceptable(check)) {
-                    // we can not have in our config test for all tokens
-                    continue;
-                }
-
-                Set<String> configTokens = configCheckTokens.get(checkName);
+            if (instance instanceof AbstractCheck
+                    && !isAllTokensAcceptable((AbstractCheck) instance)) {
+                // we can not have in our config test for all tokens
+                check = (AbstractCheck) instance;
+            }
+            else {
+                continue;
+            }
 
-                if (configTokens == null) {
-                    configTokens = new HashSet<>();
+            Set<String> configTokens = configCheckTokens.get(checkName);
 
-                    configCheckTokens.put(checkName, configTokens);
+            if (configTokens == null) {
```

Comment:
Please refactor this with `configCheckTokens.computeIfAbsent`

Reply:
Can't be done, `configTokens` are not final nor can be made final.

Reply:
Something like that:
```java
            Set<String> configTokens = configCheckTokens.computeIfAbsent(checkName, key -> {
                Set<String> newTokens = new HashSet<>();

                configCheckTokens.put(key, newTokens);

                // add all overridden tokens
                final Set<String> overrideTokens = tokensToIgnore.get(key);

                if (overrideTokens != null) {
                    newTokens.addAll(overrideTokens);
                }

                newTokens.addAll(CheckUtil.getTokenNameSet(check.getRequiredTokens()));
                checkTokens.put(key, CheckUtil.getTokenNameSet(check.getAcceptableTokens()));
                return newTokens;
            });
```
But I'm OK to keep your version.